### PR TITLE
add draft to guidelines from slack discussion

### PR DESCRIPTION
**Pull request by lpantano**
Following slack discussion we agree on adding some text on PR review to make it more transparent to new people or as a reminder.

@netlify /docs/tutorials/contributing_to_nf-core/contributing_to_pipelines

### CODE REVIEW

File: sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md [hunk 1]
```diff
@@ -80,6 +81,20 @@ Only in the unlikely and regretful event of a release happening with a bug.
 - Fix the bug, and bump version (X.Y.Z+1).
 - A PR should be made on `master` from patch to directly resolve this particular bug.
 
+## PR review guidelines
+
+- Authors/Reviewers can request help for review at nf-core Slack `#request-review` channel.
+- Anyone can merge after one positive review and with no obvious open questions and, merge ASAP so as not to leave approved PRs hanging.
+- Exception in merges to master for releases (two reviews).
```
Comment (ewels) [inline sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md:88]: Indent these as they're sub-items of the above

File: sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md [hunk 2]
```diff
@@ -80,6 +81,20 @@ Only in the unlikely and regretful event of a release happening with a bug.
...
+- Exception in pseudo PR review for first release (needs review from core or maintainer team).
```
Comment (ewels) [inline sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md:89]: Link to docs for first release

File: sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md [hunk 3]
```diff
@@ -80,6 +81,20 @@ Only in the unlikely and regretful event of a release happening with a bug.
...
+- If it's a major change then it's ok to request a second review - either the author or reviewer can do this.
+- PRs that have approving reviews but open questions can be merged by the author after addressing the questions to a common-sense level of satisfaction.
+- If the reviewer feels uneasy about this then they shouldn't leave an approval, but rather a request-changes.
+- We prefer to use merge commits to merge. This is to avoid merge conflicts when multiple people are pulling with overlapping feature branches.
+- We prefer verbose commit histories but easy merges.
```
Comment (ewels) [inline sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md:94]: Indent as it's an explanation

File: sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md [hunk 4]
```diff
@@ -80,6 +81,20 @@ Only in the unlikely and regretful event of a release happening with a bug.
...
+- Feature branches should be immediately deleted after merge. Ideally by selecting the new feature on GitHub repo settings
+- Note: squash merging commits in a PR and merging after is fine, that is up to the individuals.
```
Comment (ewels) [inline sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md:96]: Squash merging -> squashing 

Maybe needs a bit more explanation and is a sub point of the merge commit one above

**Review by ewels — COMMENTED** [review]
Needs a little polishing yet I think, left a few comments. Nothing major though.

What do folks think about having this as a new dedicated _Guidelines_ page and linking to it from here?

I had envisioned it more as a set of rules / standards that we follow rather than instructions in a tutorial tbh..

**Comment by ewels** [comment]
Also maybe some mention that it's fine / encouraged for other people to push to your PR. For example fixing linting (can mention bot commands) but also to build on or fix code if appropriate.

File: sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md (hunk 1 above)
Comment (jfy133) [reply sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md:88]:
```suggestion
- Exception: merges to master for releases requires **two** reviews.
```

I think the grammar might confuse some people

File: sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md (hunk 2 above)
Comment (jfy133) [inline sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md:89]:
```suggestion
- Exception: pseudo PR reviews for a first release needs **review from core or maintainer team**.
```

File: sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md [hunk 5]
```diff
@@ -80,6 +81,20 @@ Only in the unlikely and regretful event of a release happening with a bug.
 - Fix the bug, and bump version (X.Y.Z+1).
 - A PR should be made on `master` from patch to directly resolve this particular bug.
 
+## PR review guidelines
+
+- Authors/Reviewers can request help for review at nf-core Slack `#request-review` channel.
+- Anyone can merge after one positive review and with no obvious open questions and, merge ASAP so as not to leave approved PRs hanging.
+- Exception in merges to master for releases (two reviews).
+- Exception in pseudo PR review for first release (needs review from core or maintainer team).
+- If it's a major change then it's ok to request a second review - either the author or reviewer can do this.
```
Comment (jfy133) [inline sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md:90]:
```suggestion
- If it's there is a major change in any type of PR (from modules to pipelines) then it's ok to request a second review - either the author or reviewer can do this.
```

File: sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md [hunk 6]
```diff
@@ -80,6 +81,20 @@ Only in the unlikely and regretful event of a release happening with a bug.
...
+- PRs that have approving reviews but open questions can be merged by the author after addressing the questions to a common-sense level of satisfaction.
```
Comment (jfy133) [inline sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md:91]:
```suggestion
- PRs that have approving reviews but remaining open questions can be merged by the author after addressing the questions to a common-sense level of satisfaction.
```

**Review by jfy133 — COMMENTED** [review]
Couple of prhasing tweaks but overall LGTM after the indentation from Phil (agree reads a bit weird as they are explanation)

File: sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md (hunk 2 above)
Comment (lpantano) [reply sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md]: sorry, I cannot find exactly this? I found pipeline release but not first release

**Comment by lpantano** [comment]
thanks! I made a bunch of changes following the comments, much better now, not sure if I am 100% in the indents of all the points, let me know. 
Agree to have it outside tutorial, are you thinking Guidelines/Pipelines?

**Comment by lpantano** [comment]
or just under Guidelines?

**Comment by lpantano** [comment]
@ewels and @jfy133 , I think this is almost there. Just need to figure out the final place for this. Is inside Guidelines a good place?

File: sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md [hunk 7]
```diff
@@ -80,6 +81,22 @@ Only in the unlikely and regretful event of a release happening with a bug.
 - Fix the bug, and bump version (X.Y.Z+1).
 - A PR should be made on `master` from patch to directly resolve this particular bug.
 
+## PR review guidelines
+
+- Authors/Reviewers can request help for review at nf-core Slack `#request-review` channel.
+- We encourage collaboration in PR, other contributors are welcome to fix linting errors (you can mention relevant bot commands) or make improvements to the code if necessary.
+- Authors/Reviewers should merge after one positive review and with no obvious open questions.
+  - Exception: merges to master for releases requires **two** reviews.
+  - Exception: pseudo PR reviews for a first release needs **review from core or maintainer team**.
```
Comment (jfy133) [inline sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md:90]:
```suggestion
  - Exception: pseudo PR reviews for a first release needs **at least one review from core or maintainer team**.
```

File: sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md [hunk 8]
```diff
@@ -80,6 +81,22 @@ Only in the unlikely and regretful event of a release happening with a bug.
...
+  - If it's there is a major change in any type of PR (from modules to pipelines) then it's ok to request a second review - either the author or reviewer can do this.
```
Comment (jfy133) [inline sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md:91]:
```suggestion
  - If there is a major change in any type of PR (from modules to pipelines) then it's ok to request a second review - either the author or reviewer can do this.
```

File: sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md [hunk 9]
```diff
@@ -80,6 +81,22 @@ Only in the unlikely and regretful event of a release happening with a bug.
...
+  - PRs that have approving reviews but remaining open questions can be merged by the author after addressing the questions to a common-sense level of satisfaction.
+  - If the reviewer feels uneasy about this then they shouldn't leave an approval, but rather a request-changes.
```
Comment (jfy133) [inline sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md:93]:
```suggestion
  - If the reviewer feels uneasy about a remaining open question then they shouldn't leave a only a comment or approval, but rather a request-changes.
```

File: sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md [hunk 10]
```diff
@@ -80,6 +81,22 @@ Only in the unlikely and regretful event of a release happening with a bug.
...
+  - **request changes** reviews can be [dismissed](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/dismissing-a-pull-request-review) by the PR author if considered out of date and resolved.
```
Comment (jfy133) [inline sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md:94]:
```suggestion
  - **request changes** reviews can be [dismissed](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/dismissing-a-pull-request-review) by the PR author if considered out of date and resolved.
  - If a PR has comments from one reviewer with open questions, and requests to re-review but does not perform the re-review, the PR may be merged if the first reviewer does not perform the review after 3 months if the PR gets an independent approval from someone else.
```

Something that was discussed in the last maintainers meeting

**Review by jfy133 — COMMENTED** [review]
Ugh sorry @lpantano ! The first re-review got buried in my notifications :see_no_evil: 

Re the location: Reading through it I think a guidelines page would make sense - you use a lot of 'SHOULD' language which would fit, but if we do move there I feel we need to tweak the language to follow e.g. the modules guidelines (with a strict definition of SHOULD/MUST/MUST NOT) etc.

Each major point in this list should be become it's own header which we can then number using that astro plugin @mashehu has added to the top of e.g. the modules guidelines, and we can 'cite' them.

**Comment by jfy133** [comment]
@nf-core-bot fix linting

File: sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md [hunk 11]
```diff
@@ -42,7 +43,7 @@ If you'd like to write some code for an nf-core pipeline, the standard workflow
 2. [Fork](https://help.github.com/en/github/getting-started-with-github/fork-a-repo) the pipeline repository to your GitHub account
 3. Make the necessary changes / additions on `dev` branch by using [git checkout](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork) within your forked repository following [pipeline conventions](#pipeline-contribution-conventions)
 4. Use `nf-core pipelines schema build` and add any new parameters to the pipeline JSON schema (requires [nf-core tools](https://github.com/nf-core/tools) >= 1.10).
-5. Submit a Pull Request against the `dev` branch and wait for the code to be reviewed and merged
+5. Submit a Pull Request against the `dev` branch and wait for the code to be reviewed and merged. See below guidelines for this.
```
Comment (jfy133) [inline sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md:46]: I guess this should be changed to point to theother file?

**Comment by lpantano** [comment]
@nf-core-bot fix linting

**Comment by lpantano** [comment]
ok, got it, I think links are working, fix some typos, fix links in the tutorial section. The only think missing is:
-> Link to docs for first release https://github.com/nf-core/website/pull/2780#discussion_r1793939484
I don't know where is that...

File: sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md (hunk 2 above)
Comment (mashehu) [reply sites/docs/src/content/docs/tutorials/contributing_to_nf-core/contributing_to_pipelines.md]: https://nf-co.re/docs/tutorials/adding_a_pipeline/first_release

**Review by jfy133 — APPROVED** [review]
:tada: :tada:

**Comment by lpantano** [comment]
@jfy133 I don't think I can merge.

**Pull request merged**